scripts/SANS/sans/state/automatic_setters.py

```diff
@@ -43,7 +42,11 @@ def forwarding_setter(value, builder_instance, attribute_name_list):
 def update_the_method(builder_instance,  new_methods, setter_name, attribute_name, attribute_name_list):
     setter_name_copy = list(setter_name)
     setter_name_copy.append(attribute_name)
-    method_name = "_".join(setter_name_copy)
+    try:
+        method_name = "_".join(setter_name_copy)
+    except TypeError:
+        # An enum is being used for a key - the dev needs to create a manual setter
```

Comment:
Do we want this to fail silently here?

Reply:
Originally, yes as this would have warned all the time

However, there was an issue with using enums as keys (can't remember exactly) so we can now throw again

Long term, these automatic setters need to be removed with the extra builder patterns